### ISSUE TITLE
Auto create tables

### ISSUE THREAD

**Issue by lorensr** [Slava/meteor-rethinkdb#4]
Would be nice to automatically create tables that are declared but not present in db? 

![image](https://cloud.githubusercontent.com/assets/251288/7438799/540a8e9a-f01f-11e4-943d-b17d434f0aa4.png)

It will trip people up, even if explained in readme

**Comment by Slava** [comment]
The rethinkdb driver doesn't autocreate tables. Why would Meteor do this?

**Comment by lorensr** [comment]
1. To match the behavior of making Meteor Mongo.Collections
2. Making things easier for developer

**Comment by Slava** [comment]
Yeah, I don't know if this is what RethinkDB developers would expect. We can certainly do this, but it is kinda weird given that Rethink has an explicit command for creating tables.

**Comment by lorensr** [comment]
They might not expect it, but if they're declaring the table and it doesn't
exist, they want it to exist, and so we're doing them a favor. IMO a net
good

**Comment by Slava** [comment]
Or you misspelled the name of the table and now you have two tables. In mongo at least the collection is not created if nothing was written to it.

**Comment by just-be-dev** [comment]
Please add these to [neumino/reglite#3](https://github.com/neumino/reqlite/issues/3#issuecomment-98271848).

**Comment by smeijer** [comment]
I might hope that developers will detect "misspelled table names" while they're still actively writing their code. It is unlikely that unwanted tables will be created in a production environment.

If the final word about this is indeed to not auto-create tables, I would suggest adding an option or method to make the code create the table, and not require a db-admin to  dive into the database administration interface.

Something like 

``` js
new Rethink.Table('players', { autocreate: true });
```

Or, inspired by the mongo  `_ensureIndex` method:

``` js
Players = new Rethink.Table('players');
Players.ensureTableExists(true);
```

**Comment by ciwolsey** [comment]
Agree with above post. It's no fun when you have a ton of tables and every time you deploy your app you have to create them all manually.

**Comment by Slava** [comment]
I think an `autocreate` flag might not be a bad idea. I just don't want to introduce any default behavior that is not expected from a raw RethinkDB driver. Especially, if it is modeled after MongoDB.

**Comment by rclai** [comment]
I don't know if this should be in a separate issue, but should we also account for custom primary key field names? In RethinkDB when you create a table, there's an option to set the name of the auto-generated primary key, which if not set, defaults to `id`. This is different from MongoDB, which makes `_id` the only primary key.
